### PR TITLE
V1.0-beta1

### PR DESCRIPTION
**Pull request by scisco**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
     -   [Deployment](#deployment)
         -   [create](#create)
         -   [update](#update)
+        -   [upsert](#upsert)
+        -   [Differnet deployment configurations](#differnet-deployment-configurations)
```
Comment (sethvincent) [inline docs/README.md]: Typos: 
- Differnet -> Different
- #differnet -> #different

File: docs/README.md [hunk 2]
```diff
@@ -254,6 +261,38 @@ To update an existing CF stack
...
 
+### upsert
+
+To create a stack or update it if it exists
+
+```bash
+ kes cf upsert
+```
+
+### Differnet deployment configurations
```
Comment (sethvincent) [inline docs/README.md]: Typo: 
- Differnet -> Different

File: bin/readme.js [hunk 3]
```diff
@@ -226,6 +232,36 @@
  * ```bash
  *  kes cf update
  * ```
+ * ### upsert
+ * To create a stack or update it if it exists
+ * ```bash
+ *  kes cf upsert
+ * ```
+ *
+ * ### Differnet deployment configurations
```
Comment (sethvincent) [inline bin/readme.js]: Typo: 
- Differnet -> Different

**Comment by drewbo** [comment]
@sethvincent to answer your question, the stackName is now set differently for different deployments so that alone should update the name correctly.

@scisco looks good, I fixed some of the typos. The only thing I think may be in error is that `deployment` defaults to `dev` in a few places but I think (?) it should default to `default` (in both the docs and the actual code)

File: bin/cli.js [hunk 4]
```diff
@@ -68,9 +67,8 @@ const init = function () {
     // copy simple config file and template
     const config = yaml.safeLoad(fs.readFileSync(
       path.join(__dirname, '..', 'examples/lambdas/config.yml'), 'utf8'));
-    config.stackName = result.stack;
-    config.stage = result.stage;
-    config.buckets.internal = result.bucket;
+    config.default.stackName = result.stack;
+    config.default.buckets.internal = result.bucket;
```
Comment (sethvincent) [inline bin/cli.js]: When running `kes init` the `config.default.buckets` property is undefined:

```
TypeError: Cannot set property 'internal' of undefined
    at /Users/sdv/workspace/developmentseed/kes/bin/cli.js:72:37
    at /Users/sdv/workspace/developmentseed/kes/node_modules/prompt/lib/prompt.js:336:32
    at /Users/sdv/workspace/developmentseed/kes/node_modules/prompt/node_modules/utile/node_modules/async/lib/async.js:154:25
    at assembler (/Users/sdv/workspace/developmentseed/kes/node_modules/prompt/lib/prompt.js:333:9)
    at /Users/sdv/workspace/developmentseed/kes/node_modules/prompt/lib/prompt.js:342:32
    at /Users/sdv/workspace/developmentseed/kes/node_modules/prompt/lib/prompt.js:625:5
    at Interface.onLine (/Users/sdv/workspace/developmentseed/kes/node_modules/prompt/node_modules/read/lib/read.js:111:5)
    at emitOne (events.js:77:13)
```

Should we add a default `buckets` entry to that `examples/lambdas/config.yml` file?

Or something like `if (!config.default.buckets) config.default.buckets = {}`

**Comment by sethvincent** [comment]
I just noticed that when running `kes init` only `cloudformation.template.yml` and `config.yml` are copied.

Should it also copy default `.env` and `kes.js` files? Either that or we'll need docs updates about which files are created / what should be in them.

**Pull request merged**